src/catala/surface/lexer_fr.ml

```diff
@@ -530,24 +530,12 @@ let lex_law_fr (lexbuf : lexbuf) : token =
       else if Filename.extension name = ".pdf" then
         LAW_INCLUDE (Ast.PdfFile ((name, Pos.from_lpos pos), pages))
       else LAW_INCLUDE (Ast.CatalaFile (name, Pos.from_lpos pos))
-  | "@@", Plus (Compl '@'), "@@", Star '+' ->
-      let extract_code_title = R.regexp "@@([^@]+)@@([\\+]*)" in
-      let get_match = R.get_substring (R.exec ~rex:extract_code_title (Utf8.lexeme lexbuf)) in
-      let get_new_lines = R.regexp "\n" in
-      let new_lines_count =
-        try Array.length (R.extract ~rex:get_new_lines (Utf8.lexeme lexbuf)) with Not_found -> 0
-      in
-      for _i = 1 to new_lines_count do
-        new_line lexbuf
-      done;
-      let law_title = get_match 1 in
-      let precedence = String.length (get_match 2) in
-
-      LAW_HEADING (law_title, precedence)
-  | "@", Plus (Compl '@'), "@" ->
+  | "####", Star white_space, '[', Star white_space, Plus (Compl '#'), Star white_space, ']', '\n'
```

Comment:
So here instead of forcing four `"` we should allow for an arbitrary number, like law headings. The number of `#` should be stored in a precedence counter attached to `LAW_ARTICLE` like `LAW_HEADING`.

Reply:
All right, I understood that the predecessor of the articles was fixed.